### ISSUE TITLE
Build issue with scons 3.0.0 using Python 3

### ISSUE THREAD

**Issue by novemberist** [godotengine/godot#11860]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Arch Linux, Godot 3 master 

**Issue description:**
I know this is probably not going to become an issue for most of you soon, but just to give you a little heads-up: Godot currently fails to build with scons 3.0, which just hit ArchLinux' [testing] repo. I don't know how far off scons 3 is for other distros, but maybe there's a version independant fix for this ahead of time. This is the error I'm getting:

```
[Initial build] make_authors_header(["editor/authors.gen.h"], ["AUTHORS.md"])
scons: *** [editor/authors.gen.h] NameError : name 'unicode' is not defined
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/SCons/Action.py", line 1197, in execute
    result = self.execfunction(target=target, source=rsources, env=env)
  File "/var/aur/godot-git/src/godot/editor/SCsub", line 183, in make_authors_header
    current_section = escape_string(sections_id[i])
  File "/var/aur/godot-git/src/godot/editor/SCsub", line 10, in escape_string
    if isinstance(s, unicode):
NameError: name 'unicode' is not defined
scons: building terminated because of errors.
```

**Steps to reproduce:**

try to build godot 3 with scons 3.0

**Comment by groud** [comment]
Probably due to the python version.

**Comment by groud** [comment]
I can't reproduce on a similar distro.

**Comment by novemberist** [comment]
what are your scons and python versions?

**Comment by Noshyaar** [comment]
Related #11592

**Comment by akien-mga** [comment]
The problem is not SCons 3.0.0, but the usage of Python 3 AFAICT. #11592 introduced code which is not valid in Python 3 (cc @SaracenOne and @hoelzl)

**Comment by novemberist** [comment]
is there a way to force scons to use python 2 as a temporary fix?

**Comment by N0hbdy** [comment]
I wasn't able to force it using the hash-bang, but if you set up a virtualenv using python 2 it works (I had to do this, I'm not in front of that computer now but can recreate the steps later if you run into issues)

**Comment by hoelzl** [comment]
This issue should be fixed by PR #11843.

**Comment by akien-mga** [comment]
Fixed by #11843.

**Comment by novemberist** [comment]
This indeed seems to be fixed, unless you try to build Godot with module_mono_enabled=yes, although this might be a different issue altogether:

```
TypeError: a bytes-like object is required, not 'str':
  File "/var/aur/godot-git/src/godot/SConstruct", line 429:
    SConscript("modules/SCsub")
  File "/usr/lib/python3.6/site-packages/SCons/Script/SConscript.py", line 607:
    return method(*args, **kw)
  File "/usr/lib/python3.6/site-packages/SCons/Script/SConscript.py", line 544:
    return _SConscript(self.fs, *files, **subst_kw)
  File "/usr/lib/python3.6/site-packages/SCons/Script/SConscript.py", line 253:
    call_stack[-1].globals)
  File "/var/aur/godot-git/src/godot/modules/SCsub", line 19:
    SConscript(x + "/SCsub")
  File "/usr/lib/python3.6/site-packages/SCons/Script/SConscript.py", line 607:
    return method(*args, **kw)
  File "/usr/lib/python3.6/site-packages/SCons/Script/SConscript.py", line 544:
    return _SConscript(self.fs, *files, **subst_kw)
  File "/usr/lib/python3.6/site-packages/SCons/Script/SConscript.py", line 253:
    call_stack[-1].globals)
  File "/var/aur/godot-git/src/godot/modules/mono/SCsub", line 51:
    make_cs_files_header('glue/cs_files', 'glue/cs_compressed.gen.h')
  File "/var/aur/godot-git/src/godot/modules/mono/SCsub", line 8:
    header.write('/* This is an automatically generated file; DO NOT EDIT! OK THX */\n')

```

**Comment by hoelzl** [comment]
I haven't built Godot with Mono yet, but this error message strongly suggests another Python 2/3 incompatibility. Starting a build with Mono right now and will try to fix any bugs I encounter.

**Comment by hoelzl** [comment]
Mono build should be fixed in PR #11910 (at least the Python 3 part).
